cuequivariance_torch/cuequivariance_torch/primitives/segmented_polynomial.py

```diff
@@ -85,6 +88,89 @@ def tensor_product_uniform_1d_jit(
     tensor_product_uniform_1d_jit = None
 
 
+class SegmentedPolynomialNaive(nn.Module):
+    def __init__(
+        self,
+        polynomial: cue.SegmentedPolynomial,
+        math_dtype: torch.dtype = torch.float32,
+        output_dtype_map: List[int] = None,
+        name: str = "segmented_polynomial",
+    ):
+        super().__init__()
+
+        # TODO: Add a bunch of checks?
+
+        self.num_inputs = polynomial.num_inputs
+        self.num_outputs = polynomial.num_outputs
+        self.name = name
+        self.math_dtype = math_dtype
+        self.out_size = [o.size for o in polynomial.outputs]
+        default_dtype_map = [
+            0 if polynomial.num_inputs >= 1 else -1
+        ] * polynomial.num_outputs
+        self.dtypes = list(range(self.num_inputs)) + (
+            default_dtype_map if output_dtype_map is None else output_dtype_map
+        )
+        
+        self.graphs = []
+        self.b_outs = []
+        self.input_inds = []
+        for operation, d in polynomial.operations:
+            ope_out, b_out = operation.output_operand_buffer(self.num_inputs)
+            self.graphs.append(_tensor_product_fx(
+                d.move_operand_last(ope_out),
+                device="cuda", # TODO: Handle this?
+                math_dtype=self.math_dtype,
+                optimize_einsums=True, # TODO: Handle this?
+            ))
+            self.b_outs.append(b_out-self.num_inputs)
+            self.input_inds.append(operation.input_buffers(self.num_inputs))
+
+    def forward(
+        self,
+        inputs: List[torch.Tensor],
+        input_indices: Optional[Dict[int, torch.Tensor]] = None,
+        output_shapes: Optional[Dict[int, torch.Tensor]] = None,
+        output_indices: Optional[Dict[int, torch.Tensor]] = None,
+    ):
+        # TODO: more checks
+        
+        # Input indexing
+        for k, v in input_indices.items():
+            inputs[k] = inputs[k][v]
+
+        # Output indices:
+        out_indices = [None for _ in range(self.num_outputs)]
+        for k, v in output_indices.items():
+            out_indices[k] = v
+
+        # Check: we're always assuming a single batch axis
+        batch_dim = max(t.shape[0] for t in inputs)
```

Comment:
be careful with the case of batch size 0. If the input shape[0] are [0, 0, 1] then the batch_dim should be 0 (not 1)